include/alpaka/atomic/AtomicOaccExtended.hpp

```diff
@@ -114,7 +119,12 @@ namespace alpaka
                 T const& value) -> T
             {
                 return detail::criticalOp(
-                    [&]() {
+                    [&
+#    if BOOST_COMP_PGI
```

Comment:
Why do we not always capture the ptr by value to avoid the `#if` but keep the comment that nobody removed the capturing?

Reply:
I hope, that this bug gets fixed some time. Once there is a compiler version without this bug, and hopefully some others, we can move the CI to that version, make it the minimum supported version of NVHPC and remove the workaround. At the moment I would consider NVHPC OpenACC, like all offloading compilers, work in progress.